enhancements/cluster-logging/forwarder-to-otlp.md

```diff
@@ -0,0 +1,201 @@
+---
+title: forward-to-otlp
+authors:
+  - "@jcantrill"
+reviewers:
+  - "@alanconway"
+  - "@cahartma"
+  - "@pavolloffay"
+  - "@periklis"
+  - "@xperimental"
+approvers:
+  - "@alanconway"
+api-approvers: 
+  - "@alanconway"
+creation-date: 2024-04-30
+last-updated: 2024-04-30
+tracking-link:
+  - "https://issues.redhat.com/browse/LOG-4225"
+see-also:
+  - "/enhancements//cluster-logging/cluster-logging-v2-apis.md"
+replaces: []
+superseded-by: []
+---
+
+# Log Forwarder to OTLP endpoint
+
+Spec **ClusterLogForwarder.obervability.openshift/io** to forward logs to an **OTLP** endpoint
+
+## Summary
+
+The enhancement defines the modifications to the **ClusterLogForwarder** spec 
+and Red Hat log collector to allow administrators to collect and forward logs 
+to an OTLP receiver as defined by the [OpenTelemetry Observability framwork](https://opentelemetry.io/docs/what-is-opentelemetry/)
+
+## Motivation
+
+Customers continue to look for greater insight into the operational aspects of their clusters by using
+observability tools and open standards to allow them to avoid vedor lock-in.  OpenTelemetry
+
+> ... is an Observability framework and toolkit designed to create and manage telemetry 
+data such as traces, metrics, and logs. Crucially, OpenTelemetry is vendor- and tool-agnostic, meaning 
+that it can be used with a broad variety of Observability backends...as well as commercial offerings.
+
+This framework defines the [OTLP Specification](https://opentelemetry.io/docs/specs/otlp/) for vendors
+to implement.  This proposal defines the use of that protocol to specifically forward logs collected on
+a cluster.  Implementation of this proposal is the first step to embracing a community standard
+and to possibly deprecate the data model currently supported by Red Hat logging.
+
+### User Stories
+
+* As an administrator, I want to forward logs to an OTLP enabled, Red Hat managed LokiStack
+so that I can aggregate logs in my existing Red Hat manged log storage solution
+
+* As an administrator, I want to forward logs to an OTLP receiver
+so that I can use my observability tools to evaluate all my signals (e.g. logs, traces, metrics)
+
+### Goals
+
+* Implement OTLP over HTTP in the log collector to forward to any receiver that implements OTLP
+* Deprecate the [ViaQ Data Model](https://github.com/openshift/cluster-logging-operator/blob/release-5.9/docs/reference/datamodels/viaq/v1.adoc) in
+favor of [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/)
+* Allow customer to continue to use **ClusterLogForwarder** as-is in cases where they are not ready or not interested in OpenTelementry
+
+### Non-Goals
+
+* Replace the existing log collection agent with the [OTEL Collector](https://opentelemetry.io/docs/collector/)
+* Utilize OpenTelemetry Semantic Conventions when writing to other `Output` types
```

Comment:
Does the goal `Deprecate the [ViaQ Data Model]`  have implications on other output types?

Reply:
In our next API doc I propose a "v2" of viaq with the goals to:
* shrink message bloat
* have a well-defined data model up-front in lieu of the ad-hoc nature of v1.

I really see it more of a transition to OTEL but maybe the mention of it here is orthogonal to this enhancement.  I think until we make something available to customers and we see where OTEL is going in a larger sense, we can't determine the viability of ViaQ or what it means for other outputs.  It may be we need to enable an OTLP protocol on the other outputs.  Should we drop this statement given?

Reply:
My guess is we'll support OTEL and Viaq data models side by side for some time.

I don't think anybody will ever want to receive Viaq-model over OTLP - it's possible in principle, but if the user has chosen the OTLP protcol it's very unlikely they're interested in non-OTLP data models.

The other way around is more plausible -  OTEL data model over non-OTEL protocols - if people want to use the OTEL data model for unified labelling, but they can't use the OTLP protocol for some reason. Imaginary example: IBM's Qradar project wants to switch to the OTEL model, but still needs to deliver some of the log data using their entrenched rsyslog infrastructure.

I think we will start small with the simplest case  (OTEL model over OTLP http+json) and then expand based on demand. Otherwise there lot of things we _could_ do that ultimately nobody might care about.

Reply:
@pavolloffay @alanconway whats the consensus regarding this statement?  What should change or be removed?

Reply:
I agree with @alanconway comment above. Starting with a limited scope  - OTLP with OTEL semantic contentions makes the most sense.

Reply:
Removed statement based upon this thread